sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -541,4 +541,21 @@ class FilterPushdownSuite extends PlanTest {
 
     comparePlans(optimized, originalQuery)
   }
+
+  test("push down project") {
```

Comment:
This name is very generic.  `push project past sort` perhaps.  please also add a test where the push down would be invalid.